### PR TITLE
Fix bugs for m3 evaluation.

### PR DESCRIPTION
**Pull request by jiayu-google**

### CODE REVIEW

File: src/simulator/halo_simulator.py [hunk 1]
```diff
@@ -210,8 +210,7 @@ def simulated_reach_by_spend(
 
         # convert result to a ReachPoint
         impressions = self._data_set.impressions_by_spend(spends)
-        kplus_reaches = ReachPoint.frequencies_to_kplus_reaches(frequencies)
-        return ReachPoint(impressions, kplus_reaches, spends)
+        return ReachPoint(impressions, frequencies, spends)
```
Comment (riemanli) [inline src/simulator/halo_simulator.py:213]: I think we should rename `frequencies` to `kplus_reaches` since `kplus_reaches` is what `ReachPoint` needs.

Comment (riemanli) [inline src/simulator/halo_simulator.py:213]: I think we should rename `frequencies` to `kplus_reaches` since `kplus_reaches` is what `ReachPoint` needs.

**Review by riemanli — CHANGES_REQUESTED** [review]

File: src/simulator/halo_simulator.py (hunk 1 above)
Comment (jiayu-google) [reply src/simulator/halo_simulator.py:213]: Good point!

**Review by matthewclegg — APPROVED** [review]
Thank you for finding and fixing these bugs!

File: src/simulator/halo_simulator.py (hunk 1 above)
Comment (riemanli) [reply src/simulator/halo_simulator.py]:
```python
frequencies = [
            round(x) for x in estimator.estimate_cardinality(combined_sketch)
        ]
```
Could we rename the variable above in line 197 to `kplus_reaches`? I.e.
```python
kplus_reaches = [
            round(x) for x in estimator.estimate_cardinality(combined_sketch)
        ]
```
The thing essentially is `kplus_reaches`. Using `frequencies` is misleading.

Comment (jiayu-google) [reply src/simulator/halo_simulator.py]: Good point!

**Review by riemanli — APPROVED** [review]
LGTM!

**Comment by riemanli** [comment]
Hi @jiayu-google, can you check if the PR can pass all unit tests? I got two unit tests failed on my end:

1. `ExperimentalTrialTest.test_evaluate_when_there_is_a_modeling_exception`

        FAIL: test_evaluate_when_there_is_a_modeling_exception (__main__.ExperimentalTrialTest)
        ExperimentalTrialTest.test_evaluate_when_there_is_a_modeling_exception
        
        Traceback (most recent call last):
          File "./driver/tests/experimental_trial_test.py", line 348, in test_evaluate_when_there_is_a_modeling_exception
            self.assertEqual(
        AssertionError:
        - Cannot fit Goerg model when impressions <= reach.
        + Cannot fit Goerg model when impressions=reach

2. `TestPointAggregatorTest.test_mean_squared_relative_error`

        FAIL: test_mean_squared_relative_error (__main__.TestPointAggregatorTest)
        TestPointAggregatorTest.test_mean_squared_relative_error
        ----------------------------------------------------------------------
        Traceback (most recent call last):
          File "./driver/tests/test_point_aggregator_test.py", line 83, in test_mean_squared_relative_error
            self.assertEqual(
        AssertionError: 0.004189814814814815 != 1.361111111111111

**Comment by jiayu-google** [comment]
@riemanli 

Thank you so much for the careful checking! Just fixed.

**Review by riemanli — APPROVED** [review]

**Comment by RonPinelliMRC** [comment]
Ask that this be added to Summary of Pending Pull Requests for Planning Evaluation Framework https://docs.google.com/document/d/1tpplkQ9HugNLvjd6XkiMYbeaLefuGOuZc_1QPj6BZWg/edit#heading=h.bjct4c5af9wz

**Comment by jiayu-google** [comment]
@RonPinelliMRC  Thank you very much Ron.  Just added this PR and PR 58 to the summary: https://docs.google.com/document/d/1tpplkQ9HugNLvjd6XkiMYbeaLefuGOuZc_1QPj6BZWg/edit#
ptal.

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**